src/jingle_sip/jingle_sip_callbacks.erl

```diff
@@ -0,0 +1,276 @@
+
+        _ ->
+            ok
+    end,
+    noreply;
+sip_dialog_update(stop, Dialog, _) ->
+    {ok, CallID} = nksip_dialog:call_id(Dialog),
+
+    noreply;
+sip_dialog_update(_, _, _) ->
+    noreply.
+
+%% @doc
+%% This function is called for every response to the SIP INVITE
+%% SIP response contains the same headers as request
+%% That's why we need to switch `from` and `to` when preparing and routing Jingle
+%% to the request originator
+%% interpreted status codes:
+%% * 180 and 183 - provisional respons - we can send `ringing` session-info
+%% * 200 - the invite was accepted we can sent `session-accepted` stanza
+%% * 487 - this is to confirm INVITE cancelation from the other side (no action in this case)
+%% * 603 - used to decline the INVITE by the reciving side
+%% * all error responses between 400 and 700 result in genering session-terminate reason
+invite_resp_callback({resp, 180, SIPMsg, _Call}) ->
+    send_ringing_session_info(SIPMsg);
+invite_resp_callback({resp, 183, SIPMsg, _Call}) ->
+    send_ringing_session_info(SIPMsg);
+invite_resp_callback({resp, 200, SIPMsg, _Call}) ->
+    {ToJID, ToBinary} = get_user_from_sip_msg(from, SIPMsg),
+    {FromJID, FromBinary} = get_user_from_sip_msg(to, SIPMsg),
+
+    Body = nksip_sipmsg:meta(body, SIPMsg),
+    CallID = nksip_sipmsg:header(<<"call-id">>, SIPMsg),
+    {CodecMap, SDP} = nksip_sdp_util:extract_codec_map(Body),
+    OtherEls = sip_to_jingle:parse_sdp_attributes(SDP#sdp.attributes),
+
+
+    ContentEls = [sip_to_jingle:sdp_media_to_content_el(Media, CodecMap) || Media <- SDP#sdp.medias],
+
+    JingleEl = jingle_sip_helper:jingle_element(CallID, <<"session-accept">>, ContentEls ++ OtherEls),
+    IQEl = jingle_sip_helper:jingle_iq(ToBinary, FromBinary, JingleEl),
+    Acc = mongoose_acc:from_element(IQEl, FromJID, ToJID),
+    ok = mod_jingle_sip_backend:set_outgoing_accepted(CallID),
+    maybe_route_to_all_sessions(FromJID, ToJID, Acc, IQEl),
+    ok;
+invite_resp_callback({resp, 487, _SIPMsg, _Call}) ->
+    %% this error response only confirms that that the transaction was canceled
+    %% the real `session-terminate` stanza is sent by `sip_cancel/3` callback
+    ok;
+invite_resp_callback({resp, 486, SIPMsg, _Call}) ->
+    {ToJID, ToBinary} = get_user_from_sip_msg(from, SIPMsg),
+    {FromJID, FromBinary} = get_user_from_sip_msg(to, SIPMsg),
+    CallID = nksip_sipmsg:header(<<"call-id">>, SIPMsg),
+
+    ReasonEl = #xmlel{name = <<"reason">>,
+                      children = [#xmlel{name = <<"decline">>}]},
+    JingleEl = jingle_sip_helper:jingle_element(CallID, <<"session-terminate">>, [ReasonEl]),
+    IQEl = jingle_sip_helper:jingle_iq(ToBinary, FromBinary, JingleEl),
+    Acc = mongoose_acc:from_element(IQEl, FromJID, ToJID),
+    maybe_route_to_all_sessions(FromJID, ToJID, Acc, IQEl),
+    ok;
+invite_resp_callback({resp, ErrorCode, SIPMsg, _Call}) when ErrorCode >= 400, ErrorCode =< 600 ->
```

Comment:
`ErrorCode >= 400, ErrorCode =< 600` good to make a comment why this range.

Reply:
because the comment above says:
```erlang
all error responses between 400 and 700 result in genering session-terminate reason
```

Reply:
good catch!